spring-cloud-stream/src/main/java/org/springframework/cloud/stream/config/BindingServiceConfiguration.java

```diff
@@ -166,7 +166,9 @@ public BinderAwareChannelResolver binderAwareChannelResolver(BindingService bind
 	@Bean
 	@ConditionalOnProperty("spring.cloud.stream.bindings." + ERROR_CHANNEL_NAME + ".destination")
 	public SingleBindingTargetBindable<MessageChannel> errorChannelBindable(
-			@Qualifier(IntegrationContextUtils.ERROR_CHANNEL_BEAN_NAME) PublishSubscribeChannel errorChannel) {
+			@Qualifier(IntegrationContextUtils.ERROR_CHANNEL_BEAN_NAME) PublishSubscribeChannel errorChannel,
+			CompositeMessageChannelConfigurer compositeMessageChannelConfigurer) {
+		compositeMessageChannelConfigurer.configureOutputChannel(errorChannel, ERROR_CHANNEL_NAME);
```

Comment:
Well, what @sobychacko have done here is fully logical from the binding perspective:
```
@ConditionalOnProperty("spring.cloud.stream.bindings." + ERROR_CHANNEL_NAME + ".destination")
```
what means that end-user expect any out-of-the-box conversion applied exactly for this channel.

But on the other hand I agree that it is very dangerous to apply it unconditionally because `errorChannel` is `PublishSubscribe` by default and has that famous `LoggingHandler` as one of the default subscriber.

I believe what @mbogoevici want to see here is like `BridgeHandler` as a subscriber to the `errorChannel` and a `new DirectChannel()` to bridge to and already only this new channel return for the target binding. And yes, apply any appropriate configurations, including conversion interceptor.